test/integration_test.go

```diff
@@ -74,69 +90,151 @@ func TestIntegration(t *testing.T) {
+	t.Helper()
+
+	clientRBAC := client.RbacV1()
+
+	const namespaceFoo = "foo"
+
+	// Configure global namespace
 	_, err := clientRBAC.ClusterRoles().Create(&rbac.ClusterRole{
 		ObjectMeta: meta.ObjectMeta{Name: "create-configmaps"},
 		Rules: []rbac.PolicyRule{
 			{
-				APIGroups: []string{"v1"},
+				APIGroups: []string{""},
 				Verbs:     []string{"create"},
 				Resources: []string{"configmaps"},
 			},
 		},
 	})
 	require.NoError(t, err)
 
-	_, err = clientRBAC.Roles("default").Create(&rbac.Role{
+	_, err = clientRBAC.ClusterRoles().Create(&rbac.ClusterRole{
+		ObjectMeta: meta.ObjectMeta{Name: "get-logs"},
+		Rules: []rbac.PolicyRule{
+			{
+				Verbs:           []string{"get"},
+				NonResourceURLs: []string{"/logs"},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	_, err = clientRBAC.ClusterRoleBindings().Create(&rbac.ClusterRoleBinding{
+		ObjectMeta: meta.ObjectMeta{Name: "bob-can-get-logs"},
+		RoleRef: rbac.RoleRef{
+			Name: "get-logs",
+			Kind: cmd.ClusterRoleKind,
+		},
+		Subjects: []rbac.Subject{
+			{Kind: rbac.UserKind, Name: "Bob"},
+		},
+	})
+	require.NoError(t, err)
+
+	// Configure default namespace
+	_, err = clientRBAC.Roles(core.NamespaceDefault).Create(&rbac.Role{
 		ObjectMeta: meta.ObjectMeta{Name: "create-configmaps"},
 		Rules: []rbac.PolicyRule{
 			{
-				APIGroups: []string{"v1"},
+				APIGroups: []string{""},
 				Verbs:     []string{"create"},
 				Resources: []string{"configmaps"},
 			},
 		},
 	})
 	require.NoError(t, err)
 
-	_, err = clientRBAC.RoleBindings("default").Create(&rbac.RoleBinding{
+	_, err = clientRBAC.RoleBindings(core.NamespaceDefault).Create(&rbac.RoleBinding{
 		ObjectMeta: meta.ObjectMeta{Name: "alice-can-create-configmaps"},
-		RoleRef:    rbac.RoleRef{Name: "create-configmaps", Kind: "Role"},
+		RoleRef: rbac.RoleRef{
+			Name: "create-configmaps",
+			Kind: cmd.RoleKind,
+		},
 		Subjects: []rbac.Subject{
-			{Name: "Alice", Kind: "User"},
+			{Kind: rbac.UserKind, Name: "Alice"},
 		},
 	})
 	require.NoError(t, err)
 
-	_, err = clientRBAC.RoleBindings("default").Create(&rbac.RoleBinding{
+	_, err = clientRBAC.RoleBindings(core.NamespaceDefault).Create(&rbac.RoleBinding{
 		ObjectMeta: meta.ObjectMeta{Name: "rory-can-create-configmaps"},
-		RoleRef:    rbac.RoleRef{Name: "create-configmaps", Kind: "ClusterRole"},
+		RoleRef: rbac.RoleRef{
+			Name: "create-configmaps",
+			Kind: cmd.ClusterRoleKind,
+		},
 		Subjects: []rbac.Subject{
-			{Name: "Rory", Kind: "User"},
+			{Kind: rbac.UserKind, Name: "Rory"},
 		},
 	})
 	require.NoError(t, err)
 
-	_, err = clientRBAC.ClusterRoles().Create(&rbac.ClusterRole{
-		ObjectMeta: meta.ObjectMeta{Name: "get-logs"},
+	_, err = clientRBAC.Roles(core.NamespaceDefault).Create(&rbac.Role{
+		ObjectMeta: meta.ObjectMeta{Name: "scale-workloads"},
 		Rules: []rbac.PolicyRule{
 			{
-				Verbs:           []string{"get"},
-				NonResourceURLs: []string{"/logs"},
+				APIGroups: []string{""},
```

Comment:
If we wanted to support CRDs, would we need to allow the user to specify the APIGroup?

Reply:
That's a good point. We do not currently resolve APIGroup:

```
$ kubectl auth can-i get deployments.extensions
yes

$ kubectl who-can get deployments.extensions
Error: resolving resource: the server doesn't have a resource type "deployments.extensions"
```